docs/madr/decisions/073-spiffe-compliance.md

```diff
@@ -0,0 +1,136 @@
+
+### As a user, I want to have a SPIFFE-compliant SAN in a non-Kubernetes environment
+
+It would be beneficial to support connectivity between different environments. Unfortunately, our services currently lack proper identity on Universal, which makes this challenging.
+
+### As a user, I want to federate different trust domains using the Federation API exposed by the control plane
+
+It should be possible to federate different trust domains without creating additional resources, only by using the federation endpoint exposed by the control plane. This would be a valuable feature, enabling interoperability between different SPIFFE-compliant systems.
+
+### As a user, I want private key to not be transfered over network
+
+The private key should not be sent from the control plane to the dataplane. This can be addressed by redesigning our secret flow: instead of pushing certificates from the control plane, the dataplane should send a Certificate Signing Request (CSR) to the control plane, which then signs the certificate. In this setup, the private key remains local and never leaves the dataplane environment. This change may require significant work and might not be feasible in the current release, but it should be targeted for a future release.
+
+### As a user, I want to use an intermediate CA instead of a root CA on the zone
+
+When a user provides a root CA on the global control plane, we could generate a separate intermediate CA for each connecting zone. This would offer a stronger security model, ensuring that if a certificate is compromised in one zone, it does not impact other zones.
+
+## Out of scope
+
+### As a user, I want to specify certificates for a specific outbound
+
+You should use MeshExternalService in this case to configure the outbound connection and provide the required certificates.
+
+### As a user, I want my dataplanes to communicate with Istio services without going through a gateway
+
+In this scenario, you should also use MeshExternalService and include the necessary certificates in the configuration to enable direct mTLS communication.
+
+### As a user, I want to use two different sources of CA (Kuma and SPIRE) at the same time
+
+This is a complex use case and can make the configuration error prone and difficult to manage.
+We want to add some limitation to avoid issues:
+* A dataplane's identity must come from a single source.
+* For trust domains, identities must be issued by one system only — either SPIRE or Kuma, never both for the same dataplane.
+* It's valid for one dataplane to get its identity from Kuma and another from SPIRE — communication between such dataplanes should still work.
+
+To enable this setup, we’ll need to establish a federation mechanism — potentially by having Kuma expose a federation endpoint.
+
+### As a user, I want to have trust domain per namespace
+
+I think we shouldn't allow higher granularity than zone or mesh.
```

Comment:
I don't understand this limitation? AFAIU it shouldn't be complicated to just run 1 dataplane on a different identity.